pkgs/applications/science/logic/klee/default.nix

```diff
@@ -30,19 +29,43 @@
 # Enable runtime asserts. Default false.
 , runtimeAsserts ? false
 
-# Extra klee-uclibc config.
+# Klee uclibc. Defaults to the bundled version.
+, kleeuClibc ? null
+
+# Extra klee-uclibc config for the default klee-uclibc.
 , extraKleeuClibcConfig ? {}
 }:
 
 let
+  # The chosen version of LLVM.
+  chosenLLVMPackages = let
+    inherit (llvmPackages.llvm) version;
+    inherit (lib.strings) versionAtLeast versionOlder;
+
+    # Upper and lower bound on the LLVM version for this version of KLEE.
+    minLLVMVersion = 11;
+    maxLLVMVersion = 16;
+  in
+  if versionAtLeast version  (toString minLLVMVersion)
+     && versionOlder version (toString (maxLLVMVersion + 1)) then
+    llvmPackages
+  else
+    throw "Unsupported LLVM version: ${version} (must be in ${toString minLLVMVersion} to ${toString maxLLVMVersion})";
+
+  # The chosen version of klee-uclibc.
+  chosenKleeuClibc =
```

Comment:
Why not use an `assert` instead of making an entirely new variable?

Reply:
This actually is something else: it's defaulting it to the builtin klee-uclibc if the callPackage input is null.

Reply:
Oh, didn't quite see that.